hedera-node/src/main/java/com/hedera/services/store/contracts/AbstractLedgerWorldUpdater.java

```diff
@@ -249,6 +256,24 @@ public WorldLedgers wrappedTrackingLedgers(final SideEffectsTracker sideEffectsT
 		return withChangeObserver(trackingLedgers.wrapped(sideEffectsTracker));
 	}
 
+	protected void addCommittedRecordSourceId(final int recordSourceId, final RecordsHistorian recordsHistorian) {
+		ensureFamiliarityWith(recordsHistorian);
+		if (committedRecordSourceIds.isEmpty()) {
```

Comment:
Q: Why set it to empty if it's already empty?
Did you mean to check for null or did you mean to put something in the list?